### PR TITLE
fix(es/minifier): Make sequential inliner respect resolution order

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/6491.

---

# Investigation

This requires `inline`, `sequences` **and** `conditional`

## `obj` of `MemberExpr` is **resolved** first

So the code below is invalid, because `fragments` of fragments[...]` is undefined, even if it's reassigned

image

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
swc-bump:
 - swc_ecma_minifier

File: crates/swc_ecma_minifier/src/compress/optimize/sequences.rs [hunk 1]
```diff
@@ -1597,6 +1597,33 @@ where
                     return Ok(false);
                 }
 
+                // We can't merge into `[]` in some cases because `obj` is **resolved** before
+                // evaluating `[]`.
+                //
+                // See https://github.com/swc-project/swc/pull/6509
+
+                // We check only for an identifier because resolution is the problematic part.
+                // Evalutation order is not a problem.
```
Comment (jridgewell) [inline crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:1606]: Is this true? Wouldn't `(1, obj)[obj = {}, 'key']` hit the same problem?

Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:1606]: Oh, I was wrong. Nice catch, and thank you!

Comment (kdy1) [reply crates/swc_ecma_minifier/src/compress/optimize/sequences.rs:1606]: Nevermind, 
```js
var obj;
console.log((1, obj)[obj = {}, 'key'])
```

is expected to fail even without minification.

**Review by jridgewell — APPROVED** [review]

**Pull request merged**